Source/Private/TextExperiment/Component/ASTextLayout.m

```diff
@@ -134,26 +135,36 @@ - (instancetype)init {
   return self;
 }
 
-- (id)copyWithZone:(NSZone *)zone {
-  ASTextContainer *one = [self.class new];
+- (id)copyForced:(BOOL)forceCopy
+{
   dispatch_semaphore_wait(_lock, DISPATCH_TIME_FOREVER);
```

Comment:
Curious why we use a semaphore as lock in this class?

Reply:
It was inherited from YYText where he did that. I actually really like it for small lock scopes. It's a lot faster than pthread mutex and there is a risk of priority inversion but it's much smaller since the waiting thread sleeps instead of spinning.